### PR TITLE
fix(codec): support transinfo in attachment codec

### LINKED ISSUE #84: When dubbo-codec is enabled, metainfo does not take effect
**Describe the bug**

When dubbo-codec is enabled, metainfo does not take effect

For example, in a trace scenario, the trace context cannot be extracted from metadata

image

image

**To Reproduce**

1. enable kitex otel instrumentation
2. Check whether spans can be connected correctly

**Expected behavior**

spans can be connected correctly

**Screenshots**

See above

**Kitex version:**

v0.9.0

**Environment:**

```shell
GO111MODULE='on'
GOARCH='amd64'
GOBIN=''
GOCACHE='/Users/bytedance/Library/Caches/go-build'
GOENV='/Users/bytedance/Library/Application Support/go/env'
GOEXE=''
GOEXPERIMENT=''
GOFLAGS=''
GOHOSTARCH='amd64'
GOHOSTOS='darwin'
GOINSECURE=''
GOMODCACHE='/Users/bytedance/go/pkg/mod'
GONOPROXY='*.byted.org,*.everphoto.cn,git.smartisan.com,github.com/caicloud'
GONOSUMDB='*.byted.org,*.everphoto.cn,git.smartisan.com,github.com/caicloud'
GOOS='darwin'
GOPATH='/Users/bytedance/go'
GOPRIVATE='*.byted.org,*.everphoto.cn,git.smartisan.com,github.com/caicloud'
GOPROXY='https://go-mod-proxy.byted.org|https://goproxy.cn|https://proxy.golang.org|https://goproxy.byted.org|https://goproxy.cn|direct'
GOROOT='/usr/local/go'
GOSUMDB='sum.golang.google.cn'
GOTMPDIR=''
GOTOOLCHAIN='auto'
GOTOOLDIR='/usr/local/go/pkg/tool/darwin_amd64'
GOVCS=''
GOVERSION='go1.22.1'
GCCGO='gccgo'
GOAMD64='v1'
AR='ar'
CC='cc'
CXX='c++'
CGO_ENABLED='1'
GOMOD='/dev/null'
GOWORK=''
CGO_CFLAGS='-O2 -g'
CGO_CPPFLAGS=''
CGO_CXXFLAGS='-O2 -g'
CGO_FFLAGS='-O2 -g'
CGO_LDFLAGS='-O2 -g'
PKG_CONFIG='pkg-config'
GOGCCFLAGS='-fPIC -arch x86_64 -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -ffile-prefix-map=/var/folders/44/cllwqpdn3ws53wyrjg9mm9740000gp/T/go-build3372170917=/tmp/go-build -gno-record-gcc-switches -fno-common'
```

**Additional context**

### PR DESCRIPTION
**Pull request by CoderPoet**
#### What type of PR is this?

fix: https://github.com/kitex-contrib/codec-dubbo/issues/84

#### Check the PR title.

- [x] This PR title match the format: \<type\>(optional scope): \<description\>
- [x] The description of this PR title is user-oriented and clear enough for others to understand.
- [x] Attach the PR updating the user documentation if the current PR requires user awareness at the usage level. [User docs repo](https://github.com/cloudwego/cloudwego.github.io)

#### (Optional) Translate the PR title into Chinese.

在 attachment 编解码中支持处理 transinfo

#### (Optional) More detailed description for this PR(en: English/zh: Chinese).

en: support transinfo in attachment codec
zh(optional): 在 attachment 编解码器中支持 TransInfo

#### (Optional) Which issue(s) this PR fixes:

fix: https://github.com/kitex-contrib/codec-dubbo/issues/84

#### (optional) The PR that updates user documentation:

image

image

### CODE REVIEW

File: samples/helloworld/kitex/client/main.go [hunk 1]
```diff
@@ -8,16 +8,27 @@ import (
...
+	p := provider.NewOpenTelemetryProvider(
+		provider.WithServiceName(serviceName),
+		provider.WithExportEndpoint("localhost:4317"),
+		provider.WithInsecure(),
+	)
+	defer p.Shutdown(context.Background())
+
 	cli, err := greetservice.NewClient("helloworld",
-		client.WithHostPorts("127.0.0.1:21001"),
+		client.WithHostPorts("127.0.0.1:21000"),
```
Comment (DMwangnima) [inline samples/helloworld/kitex/client/main.go:25]: port 21001 is used to invoke dubbo-java side in samples/helloword/dubbo.
For testing, I would add this case in https://github.com/kitex-contrib/codec-dubbo-tests

Comment (CoderPoet) [reply samples/helloworld/kitex/client/main.go:25]: > port 21001 is used to invoke dubbo-java side in samples/helloword/dubbo. For testing, I would add this case in https://github.com/kitex-contrib/codec-dubbo-tests

get

Comment (CoderPoet) [reply samples/helloworld/kitex/client/main.go:25]: > port 21001 is used to invoke dubbo-java side in samples/helloword/dubbo. For testing, I would add this case in https://github.com/kitex-contrib/codec-dubbo-tests

done

**Review by DMwangnima — APPROVED** [review]
BTW, maybe we should exclude "go.mod" in _typos.toml.
image

**Comment by GuangmingLuo** [comment]
Yes, please add to _typos.toml.

**Comment by CoderPoet** [comment]
> BTW, maybe we should exclude "go.mod" in _typos.toml. image

+1

I'll see how I can change the rules

**Comment by CoderPoet** [comment]
> BTW, maybe we should exclude "go.mod" in _typos.toml. image

done

**Comment by CoderPoet** [comment]
> Yes, please add to _typos.toml.

done

File: _typos.toml [hunk 2]
```diff
@@ -1,4 +1,4 @@
 # Typo check: https://github.com/crate-ci/typos
 
 [files]
-extend-exclude = ["go.sum", "check_branch_name.sh"]
+extend-exclude = ["go.mod", ""go.sum", "check_branch_name.sh"]
```
Comment (DMwangnima) [inline _typos.toml]: There is a redundant quotation mark.

Comment (CoderPoet) [reply _typos.toml]: > There is a redundant quotation mark.

oh sorry

Comment (CoderPoet) [reply _typos.toml]: > There is a redundant quotation mark.

done

**Review by DMwangnima — APPROVED** [review]

**Comment by nobody-s07** [comment]
This PR introduces an issue where Kitex client calls Dubbo service:
WeChat82983fe0872f5e86ef832e9cbe364fe6

The corresponding Java code is as follows,  where 'attachments' is a Map with keys of type String:
image

In this PR, the attachment contains data with keys of type uint16:
```  
        //  pkg/dubbo_spec/attachment.go:  line 56 ~ line 58
	for k, v := range transInfo.TransIntInfo() {
		result[k] = v
	}
```

**Comment by nobody-s07** [comment]
Besides,the trace context was not correctly propagated to dubbo server side ( extracting trace context through opentelemetry-javaagent ):
image

I compared the 'attachments' received by the Dubbo service when called by the Dubbo client and the Kitex client, there are some differences in the keys within the 'attachments'.
called by dubbo client:
image
called by kitex client:
image

I made a local modification to the codec-dubbo, removing the prefix `RPC_TRANSIT_`, and found that the Dubbo server could correctly parse the trace context:
image
I'm not sure if directly removing the prefix(`RPC_TRANSIT_`) will introduce other issues, but currently, it seems that this prefix causes the Dubbo server to be unable to parse the trace context.

**Comment by CoderPoet** [comment]
> This PR introduces an issue where Kitex client calls Dubbo service: WeChat82983fe0872f5e86ef832e9cbe364fe6
> 
> The corresponding Java code is as follows, where 'attachments' is a Map with keys of type String: image
> 
> In this PR, the attachment contains data with keys of type uint16:
> 
> ```
>         //  pkg/dubbo_spec/attachment.go:  line 56 ~ line 58
> 	for k, v := range transInfo.TransIntInfo() {
> 		result[k] = v
> 	}
> ```

@nobody-s07 Sorry for not having enough testing before causing the issue. This PR（https://github.com/kitex-contrib/codec-dubbo/pull/90 ） has fixed the issue, you can try again

**Comment by nobody-s07** [comment]
@CoderPoet Thank you for your resolution! Now the trace context is correctly passed to the Dubbo service.

**Comment by DMwangnima** [comment]
> @CoderPoet Thank you for your resolution! Now the trace context is correctly passed to the Dubbo service.

Hi! Thank you very much for pointing this out. Would you be interested in contributing a testcase to https://github.com/kitex-contrib/codec-dubbo-tests ?

**Pull request merged**